From 4bffaecf557d82ad2144687420a6f970a0ecaaff Mon Sep 17 00:00:00 2001
From: Paulo Ewerton Gomes Fragoso <pauloewerton@lsd.ufcg.edu.br>
Date: Fri, 12 Jun 2015 17:22:03 +0000
Subject: [PATCH] Adding Angular Identity Projects Panel

This patch adds a blank Projects panel to the Identity dashboard.
Table, forms and actions will be added in subsequent patches.

Change-Id: Id940ec62ad6f956e4884d194c672a0131d22e0d6
Partially-Implements: blueprint angularize-identity-projects
---
 .../static/dashboard/identity/identity.module.js   |  3 +-
 .../static/dashboard/identity/identity.scss        |  1 +
 .../dashboard/identity/projects/projects.module.js | 33 ++++++++++++++++++++++
 .../identity/projects/projects.module.spec.js      | 24 ++++++++++++++++
 .../dashboard/identity/projects/projects.scss      |  0
 5 files changed, 60 insertions(+), 1 deletion(-)
 create mode 100644 openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.js
 create mode 100644 openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.spec.js
 create mode 100644 openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.scss

diff --git a/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.module.js b/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.module.js
index 3c881e9..521496f 100644
--- a/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.module.js
+++ b/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.module.js
@@ -26,7 +26,8 @@
    */
   angular
     .module('hz.dashboard.identity', [
-      'hz.dashboard.identity.users'
+      'hz.dashboard.identity.users',
+      'hz.dashboard.identity.projects'
     ]);
 
 })();
diff --git a/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.scss b/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.scss
index 9ad398e..b95176f 100644
--- a/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.scss
+++ b/openstack_dashboard/dashboards/identity/static/dashboard/identity/identity.scss
@@ -1,2 +1,3 @@
 
 @import "users/users";
+@import "projects/projects";
diff --git a/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.js b/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.js
new file mode 100644
index 0000000..bf513fa
--- /dev/null
+++ b/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.js
@@ -0,0 +1,33 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+(function() {
+  'use strict';
+
+  /**
+   * @ngdoc hz.dashboard.identity.projects
+   * @ng-module
+   * @description
+   * Provides all of the services and widgets required
+   * to support and display the identity projects.
+   */
+  angular
+    .module('hz.dashboard.identity.projects', [])
+    .constant('hz.dashboard.identity.projects.basePath', basePath());
+
+  function basePath() {
+    return (window.WEBROOT || '') + '/static/dashboard/identity/projects/';
+  }
+
+})();
diff --git a/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.spec.js b/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.spec.js
new file mode 100644
index 0000000..d0c5626
--- /dev/null
+++ b/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.module.spec.js
@@ -0,0 +1,24 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+(function() {
+  'use strict';
+
+  describe('Identity projects module', function() {
+    it('should exist', function() {
+      expect(angular.module('hz.dashboard.identity.projects')).toBeDefined();
+    });
+  });
+
+})();
diff --git a/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.scss b/openstack_dashboard/dashboards/identity/static/dashboard/identity/projects/projects.scss
new file mode 100644
index 0000000..e69de29
-- 
1.9.1

